src/watchdog.rs

```diff
@@ -0,0 +1,82 @@
+//! System Window Watchdog implementation
+
+use crate::rcc::Ccdr;
+use crate::time::Hertz;
+use cast::u8;
+use crate::hal::watchdog::{Watchdog, WatchdogEnable};
+use stm32h7::stm32h7x3::WWDG;
+
+/// Implements the System Window Watchdog
+pub struct SystemWindowWatchdog {
+    wwdg: WWDG,
+    down_counter: u8,
+    pclk3_frequency: Hertz,
+}
+
+impl SystemWindowWatchdog {
+    /// Returns a System Window Watchdog object with down_counter intialized to zero
+    /// to indicate the clock has not been used yet
+    pub fn new(wwdg: WWDG, ccdr: &Ccdr) -> Self {
+        // enable the peripheral inside the APB3
+        ccdr.rb.apb3enr.modify(|_, w| w.wwdg1en().set_bit());
+        SystemWindowWatchdog {
+            wwdg,
+            down_counter: 0,
+            pclk3_frequency: ccdr.clocks.pclk3(),
+        }
+    }
+}
+
+impl Watchdog for SystemWindowWatchdog {
+    /// Feeds the watchdog in order to avoid a reset, only executes properly if the watchdog
+    /// has already been started aka. the down_counter is not 0 anymore
+    fn feed(&mut self) {
+        // if this value is 0 it is assumed that the watchdog has not yet been started
+        assert!(self.down_counter != 0);
+        self.wwdg.cr.modify(|_, w| w.t().bits(self.down_counter));
+    }
+}
+
+impl WatchdogEnable for SystemWindowWatchdog {
+    type Time = u32;
+    /// Starts the watchdog with a given timeout period, if this period is out of bounds the function
+    /// is going to panic
+    fn start<T>(&mut self, period: T)
+    where
+        T: Into<Self::Time>,
+    {
```

Comment:
The units of `period` should be clear and documented.

A clear option is to use a separate type for milliseconds, defined in `time.rs`
```
/// Time unit
#[derive(PartialEq, PartialOrd, Clone, Copy)]
pub struct MilliSeconds(pub u32);
```